File: projetoJava/src/ContaCorrente.java
public class ContaCorrente {
    private double saldo;

    public ContaCorrente(double saldoInicial) {
        this.saldo = saldoInicial;
    }

    public void sacar(double valor) throws SaldoInsuficienteException {
        if (valor > saldo) {
            throw new SaldoInsuficienteException(saldo, valor);
        }
        saldo -= valor;
    }

    public void depositar(double valor) {
        saldo += valor;
    }

    public double getSaldo() {
        return saldo;
    }
}

File: projetoGo/contaCorrente.go
package main

type ContaCorrente struct {
	saldo float64
}

func NovoContaCorrente(saldoInicial float64) *ContaCorrente {
	return &ContaCorrente{saldo: saldoInicial}
}

func (c *ContaCorrente) Sacar(valor float64) error {
	if valor > c.saldo {
		return &SaldoInsuficienteException{saldoAtual: c.saldo, valorSaque: valor}
	}
	c.saldo -= valor
	return nil
}

func (c *ContaCorrente) Depositar(valor float64) {
	c.saldo += valor
}

func (c *ContaCorrente) GetSaldo() float64 {
	return c.saldo
}
